### PR TITLE
Include test for edge-ami image type

### PR DESCRIPTION
**Pull request by mcattamoredhat**
Now that nightly repo has osbuild-composer-85-1.el9.x86_64.rpm included, osbuild-composer is able to generate an edge AMI that can be uploaded and deployed on AWS. It would be useful to include a dedicated test script for automation in downstream.

ostree-ami-image.sh uses osbuild-composer to build edge-ami image type, upload it to a S3 bucket and deploy an EC2 instance using that image.

Support to aarch64 has been included.

### CODE REVIEW

**Comment by henrywang** [comment]
Could you please fix commit line error. I just added this test in CI. Thanks.

File: ostree-ami-image.sh [hunk 1]
```diff
@@ -0,0 +1,918 @@
...
+      }
+    ]
+  }
+}
+EOF
+sudo chmod +r "${IGNITION_CONFIG_SAMPLE_PATH}" "${IGNITION_CONFIG_PATH}"
+
+# Start AWS cli installation 
+if [[ "${ARCH}" == x86_64 ]]; then
+    AWS_CLI_URL="https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip"
```
Comment (henrywang) [inline ostree-ami-image.sh:476]: `https://awscli.amazonaws.com/awscli-exe-linux-${ARCH}.zip` is light.

File: ostree-ami-image.sh [hunk 2]
```diff
@@ -0,0 +1,918 @@
...
+ansible_private_key_file=${SSH_KEY}
+ansible_ssh_common_args="-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+ansible_become=yes 
+ansible_become_method=sudo
+ansible_become_pass=${EDGE_USER_PASSWORD}
+EOF
+
+# Test IoT/Edge OS
+if [[ "${ARCH}" == x86_64 ]]; then
+    podman run --annotation run.oci.keep_original_groups=1 -v "$(pwd)":/work:z -v "${TEMPDIR}":/tmp:z --rm quay.io/rhel-edge/ansible-runner:latest ansible-playbook -v -i /tmp/inventory -e os_name="${ANSIBLE_OS_NAME}" -e ignition="true" -e ostree_commit="${INSTALL_HASH}" -e ostree_ref="${REF_PREFIX}:${OSTREE_REF}" -e sysroot_ro="$SYSROOT_RO" check-ostree.yaml || RESULTS=0
```
Comment (henrywang) [inline ostree-ami-image.sh:753]: Just use `quay.io/rhel-edge/ansible-runner:${ARCH}`.

File: ostree-ami-image.sh [hunk 3]
```diff
@@ -0,0 +1,918 @@
...
+ansible_private_key_file=${SSH_KEY}
+ansible_ssh_common_args="-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+ansible_become=yes
+ansible_become_method=sudo
+ansible_become_pass=${EDGE_USER_PASSWORD}
+EOF
+
+# Test IoT/Edge OS
+if [[ "${ARCH}" == x86_64 ]]; then
+    podman run --annotation run.oci.keep_original_groups=1 -v "$(pwd)":/work:z -v "${TEMPDIR}":/tmp:z --rm quay.io/rhel-edge/ansible-runner:latest ansible-playbook -v -i /tmp/inventory -e os_name="${ANSIBLE_OS_NAME}" -e ignition="true" -e ostree_commit="${UPGRADE_HASH}" -e ostree_ref="${REF_PREFIX}:${OSTREE_REF}" -e sysroot_ro="$SYSROOT_RO" check-ostree.yaml || RESULTS=0
```
Comment (henrywang) [inline ostree-ami-image.sh:908]: Same as above.

**Review by henrywang — CHANGES_REQUESTED** [review]

File: ostree-ami-image.sh (hunk 1 above)
Comment (mcattamoredhat) [reply ostree-ami-image.sh]: Done!

File: ostree-ami-image.sh (hunk 2 above)
Comment (mcattamoredhat) [reply ostree-ami-image.sh]: Done!

File: ostree-ami-image.sh (hunk 3 above)
Comment (mcattamoredhat) [reply ostree-ami-image.sh]: Done!

**Review by henrywang — APPROVED** [review]
LGTM. Thanks.
